src/p11_child/p11_child_openssl.c

```diff
@@ -1957,220 +2182,38 @@ errno_t do_card(TALLOC_CTX *mem_ctx, struct p11_ctx *p11_ctx,
         goto done;
     }
 
-    if (slot_id == (CK_SLOT_ID)-1) {
+    if (*_multi[0] == '\0') {
         DEBUG(SSSDBG_TRACE_ALL, "Token not present.\n");
-        if (p11_ctx->wait_for_card) {
-            /* After obtaining the module's slot list (in the loop above), this
-             * call is needed to let any changes in slots take effect. */
-            rv = module->C_GetSlotList(CK_FALSE, NULL, &num_slots);
-            if (rv != CKR_OK) {
-                DEBUG(SSSDBG_OP_FAILURE, "C_GetSlotList failed [%lu][%s].\n",
-                                         rv, p11_kit_strerror(rv));
-                ret = EIO;
-                goto done;
-            }
-
-            ret = wait_for_card(module, &slot_id, &info, &token_info, uri);
-            if (ret != EOK) {
-                DEBUG(SSSDBG_OP_FAILURE, "wait_for_card failed.\n");
-                goto done;
-            }
-        } else {
+        if (!p11_ctx->wait_for_card) {
```

Comment:
I understand the intention to decrease indentation level, but it would be better to have this change in a separate patch. Just to make review easier.